### PR TITLE
FSE: remove obsolete CircleCI jobs

### PR DESCRIPTION
**Pull request by vindl**
#### Changes proposed in this Pull Request

Since we moved to GitHub actions for this and removed the old sync script that was relying on CircleCI artifacts, this job should be safe to remove now.

#### Testing instructions

The CircleCI tests should pass.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -321,18 +320,6 @@ jobs:
...
-    parallelism: 1
-    steps:
-      - prepare
-      - run:
-          name: Build the Full Site Editing plugin and theme
-          command: |
-            npx lerna run build --scope='@automattic/full-site-editing'
-            tar -czvf $CIRCLE_ARTIFACTS/full-site-editing/build-archive.tar.gz -C apps/full-site-editing/full-site-editing-plugin .
-      - store-artifacts-and-test-results
   test-full-site-editing:
```
Comment (vindl) [inline .circleci/config.yml:323]: @noahtallen do we still need this?

Comment (noahtallen) [reply .circleci/config.yml:323]: we do need test FSE, it runs our JS unit tests. Not sure if build FSE is required for that? If so, then we could just add a build step to the tests command

Comment (vindl) [reply .circleci/config.yml:323]: We'll know once the jobs are complete. 😄

Comment (vindl) [reply .circleci/config.yml:323]: > we do need test FSE, it runs our JS unit tests.

Oh ok, I wasn't sure if we had actions equivalent of that at this point.

Comment (noahtallen) [reply .circleci/config.yml:323]: nope :)

**Review by noahtallen — APPROVED** [review]
Looks good to me! I verified that the test-fse step is working and that the tests are running as expected.

**Pull request merged**